pkg/runtime/processor/subscriptions.go

```diff
@@ -0,0 +1,93 @@
+/*
+Copyright 2024 The Dapr Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package processor
+
+import (
+	"context"
+
+	subapi "github.com/dapr/dapr/pkg/apis/subscriptions/v2alpha1"
+	"github.com/dapr/dapr/utils"
+)
+
+func (p *Processor) AddPendingSubscription(ctx context.Context, subscriptions ...subapi.Subscription) bool {
+	p.lock.Lock()
+	defer p.lock.Unlock()
+	if p.shutdown.Load() {
+		return false
+	}
+
+	scopedSubs := p.scopeFilterSubscriptions(subscriptions)
+	if len(scopedSubs) == 0 {
```

Comment:
This will not load a subscription tha has no scopes, but its possible today to use a subscription with no scopes.

Reply:
The list is of subscription manifests. If after filtering subscriptions we end up with an empty list, we exit early.

Reply:
To be clear, this will return true which signals the subscription was added but it will not in effect be added to the comp store. Is that the case?

Reply:
It returns `true` indicating that processing of this resource was successful. We classify that not processing this non-scoped subscription is a happy path result.

Reply:
So returning to my original point - if we allow loading non-scoped subscriptions on init, then shouldn't this method also do so?

Reply:
I think I'm mangling my words a bit.

This function removes subscription manifests which this app is not scoped for. A subscription is scoped when the manifest has either an empty scope string slice _or_ the appID is in the scope list. This scoping behaviour is that same as on init today. After filtering the list results in an empty list of subscriptions, there are no subscriptions to process and we can exit early signalling success.

Reply:
Ok understood